docs/okeps/okep-5217-routed-ingress-l2-udn-without-ipam.md

```diff
@@ -0,0 +1,116 @@
+# OKEP-5217: Routed ingress support for L2 UDN without dynamic IP allocation
+
+* Issue: [#5217](https://github.com/ovn-org/ovn-kubernetes/issues/5217)
+
+## Problem Statement
+
+There is an increased interest for VMs in Kubernetes, which is generating the
+desire to import VMs from another virtualization platforms.
+
+VMs running in virtualization platforms are usually using static IP allocation,
+where IPs are assigned directly in the guest.
+
+One of the features most sought after by virtualization users is routed ingress
+for L2 UDN, something which - right now - OVN-Kubernetes allows using the BGP
+feature.
+
+The current gap in OVN-Kubernetes is to provide routed ingress to VMs whose IPs
+were assigned statically in the guest. This enhancement plans to bridge that
+gap.
+
+## Goals
+
+- Detect/expose the MAC for the VM to be configured in OVNK for the VM (or disable port security)
+- When creating the UDN, be able to specify the gateway IP address, so the migrated VM can keep the same default route.
+- When creating the UDN, allow excludeSubnets to be used with L2 or L3 UDNs to ensure OVNK does not use an IP address from the range that VMs have already been assigned outside of the cluster.
+- When creating the UDN, we may need a new IPAM mode to indicate to OVNK that it should reserve IPs for internal use (gateway IP potentially, and mp0 IP), but it should not auto assign IPs to pods. Note, we may not need this, but listing it just in case.
+
+## Non-Goals
+
+- `NetworkPolicies`, `MultiNetworkPolicies`, or `Services` to work, given
+OVN-Kubernetes has no awareness of the assigned IP addresses. A way to
+introspect / discover the IP address on the guest may be required to implement
+any of those features.
+
+## Follow-up goals
+
+- Specifying the IP address of the VM for OVNK to use (but not assign) via persistent IP or some other mechanism
```

Comment:
I would add this as part of the enhancement instead of follow up, it may simplify the mechanism, and make sense to tackle together with mac address.